Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -70,18 +109,121 @@ class LoginListViewController: UIViewController {
 
         KeyboardHelper.defaultHelper.addDelegate(self)
 
-        profile.logins.getAllLogins().uponQueue(dispatch_get_main_queue()) { result in
-            self.loginDataSource.cursor = result.successValue
-            self.tableView.reloadData()
+        // If we are editing the search view, use the search term query instead
+        if searchView.isEditing {
+            searchLoginsWithText(searchView.inputField.text ?? "")
+        } else {
+            activeLoginQuery = profile.logins.getAllLogins().bindQueue(dispatch_get_main_queue()) { result in
+                self.loginDataSource.cursor = result.successValue
+                self.tableView.reloadData()
+                return succeed()
+            }
+        }
+    }
+
+    func toggleDeleteBarButton() {
+        // Show delete bar button item if we have selected any items
+        if loginSelectionController.selectedCount > 0 {
+            if (navigationItem.rightBarButtonItem == nil) {
+                navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Delete", style: .Plain, target: self, action: "SELdelete")
+                navigationItem.rightBarButtonItem?.tintColor = UIColor.redColor()
+            }
+        } else {
+            navigationItem.rightBarButtonItem = nil
+        }
+    }
+
+    func toggleSelectionTitle() {
+        if loginSelectionController.selectedCount == loginDataSource.cursor?.count {
+            selectionButton.setTitle(deselectAllTitle, forState: .Normal)
+        } else {
+            selectionButton.setTitle(selectAllTitle, forState: .Normal)
         }
     }
 }
 
+// MARK: - Selectors
+extension LoginListViewController {
+
+    func SELedit() {
+        navigationItem.rightBarButtonItem = nil
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .Cancel, target: self, action: "SELcancel")
+        selectionButtonHeightConstraint?.updateOffset(UIConstants.ToolbarHeight)
+        self.view.layoutIfNeeded()
+        tableView.setEditing(true, animated: true)
+    }
+
+    func SELcancel() {
+        selectionButtonHeightConstraint?.updateOffset(0)
+        self.view.layoutIfNeeded()
+        loginSelectionController.deselectAll()
+        tableView.setEditing(false, animated: true)
+        navigationItem.leftBarButtonItem = nil
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .Edit, target: self, action: "SELedit")
+    }
+
+    func SELdelete() {
+        let deleteAlert = UIAlertController.deleteLoginAlertWithDeleteCallback({ [unowned self] _ in
+
+            // Delete here
+            let loginsToDelete = self.loginSelectionController.selectedIndexPaths.map { indexPath in
+                self.loginDataSource.loginAtIndexPath(indexPath)
+            }
+
+            let now = NSDate.now()
+            let deleteDeferreds = loginsToDelete.map { login in
+                return self.profile.logins.deleteByGUID(login.guid, deletedAt: now)
```

Comment:
Nope — `deleteByGUID` is for Sync to use. Front-end code should use `removeLoginByGUID`.

Also, those deferreds will be firing immediately, and will all end up in the database queue. If you were to do this serially, do this instead:

```
walk(loginsToDelete) { self.profile.logins.deleteByGUID($0.guid, deletedAt: now }
```

which returns a single `Success`.

(And I'd suggest generalizing that to `removeLoginsWithGUIDs` so that this can be done in a single database transaction. Let me know if you need pointers on that.)